pkg/admission/prep_downscale.go

```diff
@@ -329,3 +338,35 @@ func findDownscalesDoneMinTimeAgo(ctx context.Context, api kubernetes.Interface,
 	}
 	return nil, nil
 }
+
+// findStatefulSetWithNonUpdatedReplicas returns any statefulset that has non-updated replicas, indicating that the statefulset
+// may be in the process of being rolled.
+func findStatefulSetWithNonUpdatedReplicas(ctx context.Context, api kubernetes.Interface, namespace, rolloutGroup string) (*statefulSetDownscale, error) {
+	stsList, err := findStatefulSetsForRolloutGroup(ctx, api, namespace, rolloutGroup)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, sts := range stsList.Items {
+		status := sts.Status
+		if !(status.Replicas == status.ReadyReplicas && status.ReadyReplicas == status.UpdatedReplicas) {
```

Comment:
It might be necessary to check the status of each of the individual pods. I'd double check the logic in the controller because I believe it does something like that since the `status` of the sts lags reality sometimes.

Reply:
Good suggestions